File: Brute_Force/python/18111.py
from collections import Counter
import sys
input = sys.stdin.readline
n,m,b = map(int,input().split())
graph = []
for _ in range(n):
    graph.extend(list(map(int,input().split())))
graph_dic = Counter(graph)
min_time = 10**18*9
height = 0 

for h in range(max(graph_dic.keys()),-1,-1):
    inventory = b
    add_block = 0
    cur_time = 0
    for cur_h, cur_cnt in graph_dic.items():
        dif = cur_h - h
        if dif > 0 :
            inventory += dif*cur_cnt
        elif dif<0:
            add_block -= dif*cur_cnt
    if inventory<add_block: continue
    cur_time = (inventory-b)*2 + add_block
    if cur_time<min_time:
        min_time = cur_time
        height = h
print(min_time,height)

File: Brute_Force/javascript/18111.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const [n,m,b] = input[0].split(' ').map(Number);
let graph=[];
for (let i = 1 ; i<=n ; i++){
  graph.push(input[i].split(' ').map(Number));
}
let minTime = Infinity;
let height = 0;
for (let h = 0,i,j ; h<= 256 ; h++){
  let removeBlock = 0;
  let addBlock = 0;
  let curTime;
  for (i = 0 ; i<n ; i++){
    for (j = 0 ; j<m ; j++){
      if (graph[i][j]>h) removeBlock += graph[i][j]-h;
      if (graph[i][j]<h) addBlock += h - graph[i][j];
    }
  }
  if (removeBlock+b-addBlock <0) continue;
  curTime = removeBlock*2 + addBlock;
  if (curTime <= minTime){
    minTime = curTime;
    height = h;
  }
}
console.log(minTime,height);